aicsimageio/readers/glob_reader.py

```diff
@@ -0,0 +1,384 @@
+                # Ensure that the outer list is the number of scenes
+                if len(channel_names) != len(self.scenes):
+                    raise exceptions.ConflictingArgumentsError(
+                        f"Number of channel name lists provided does not match the "
+                        f"number of scenes found in the file. "
+                        f"Number of scenes: {len(self.scenes)}, "
+                        f"Provided channel name lists: {dim_order}"
+                    )
+            self._channel_names = channel_names
+
+        self.chunk_dims = chunk_dims
+
+        for dim in REQUIRED_CHUNK_DIMS:
+            if dim not in self.chunk_dims:
+                self.chunk_dims.append(dim)
+
+        # Safety measure / "feature"
+        self.chunk_dims = [d.upper() for d in self.chunk_dims]
+        
+        if dim_order is not None:
+            self._dim_order = dim_order
+        else:
+            self._dim_order = "".join(d for d in DEFAULT_DIMENSION_ORDER if d in self._all_files.columns or d in self.chunk_dims)
+
+        self._channel_names = channel_names
+
+        if single_file_shape is None:
+            with self._fs.open(self._path) as open_resource:
+                with TiffFile(open_resource) as tiff:
+                    if tiff.is_shaped:
+                        self._single_file_shape = tuple(
+                            tiff.shaped_metadata[0]["shape"]
+                        )
+                    elif len(tiff.series) == 1:
+                        self._single_file_shape = tiff.series[0].shape
+
+        else:
+            self._single_file_shape = single_file_shape
+
+        if len(single_file_dims) != len(self._single_file_shape):
+            raise exceptions.ConflictingArgumentsError(
+                f"Number of single file dimensions does not match the"
+                f"number of dimensions in a test file. "
+                f"Number of dimensions in file: {len(self._single_file_shape)}, "
+                f"Provided number of dimensions: {len(single_file_dims)}."
+            )
+
+        else:
+            self._single_file_dims = list(single_file_dims)
+
+        self._single_file_sizes = dict(
+            zip(self._single_file_dims, self._single_file_shape)
+        )
+        # Enforce valid image
+        if not self._is_supported_image(self._fs, self._path):
+            raise exceptions.UnsupportedFileFormatError(
+                self.__class__.__name__, self._path
+            )
+
+    @property
+    def scenes(self) -> Tuple[str, ...]:
+        if self._scenes is None:
+            self._scenes = tuple(
+                metadata_utils.generate_ome_image_id(s)
+                for s in range(self._all_files[DimensionNames.Samples].nunique())
+            )
+        return self._scenes
+
+    def _read_delayed(self) -> xr.DataArray:
+
+        scene_files = self._all_files.loc[
+            self._all_files[DimensionNames.Samples] == self.current_scene_index
```

Comment:
Yea instead of using `DimensionNames.Samples` can you make a constant, just for clarity in code.

Reply:
Or and I think this may be a better suggestion, have a parameter called "scene_glob_character" that defaults to `"S"` but can be changed to `"I"` or something.